src/write/op.rs

```diff
@@ -0,0 +1,1558 @@
+    ///
+    /// This can be passed to `set_target`.
+    #[inline]
+    pub fn next_index(&self) -> usize {
+        self.operations.len()
+    }
+
+    /// Set the target of a `DW_OP_skip` or `DW_OP_bra` operation .
+    pub fn set_target(&mut self, operation: usize, new_target: usize) {
+        debug_assert!(new_target <= self.next_index());
+        debug_assert_ne!(operation, new_target);
+        match self.operations[operation] {
+            Operation::Skip(ref mut target) | Operation::Branch(ref mut target) => {
+                *target = new_target;
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    /// Add a `DW_OP_call4` operation to the expression.
+    pub fn op_call(&mut self, entry: UnitEntryId) {
+        self.operations.push(Operation::Call(entry));
+    }
+
+    /// Add a `DW_OP_call_ref` operation to the expression.
+    pub fn op_call_ref(&mut self, entry: Reference) {
+        self.operations.push(Operation::CallRef(entry));
+    }
+
+    /// Add a `DW_OP_convert` operation to the expression.
+    ///
+    /// `base` is the DIE of the base type, or `None` for the generic type.
+    pub fn op_convert(&mut self, base: Option<UnitEntryId>) {
+        self.operations.push(Operation::Convert(base));
+    }
+
+    /// Add a `DW_OP_reinterpret` operation to the expression.
+    ///
+    /// `base` is the DIE of the base type, or `None` for the generic type.
+    pub fn op_reinterpret(&mut self, base: Option<UnitEntryId>) {
+        self.operations.push(Operation::Reinterpret(base));
+    }
+
+    /// Add a `DW_OP_entry_value operation to the expression.
+    pub fn op_entry_value(&mut self, expression: Expression) {
+        self.operations.push(Operation::EntryValue(expression));
+    }
+
+    /// Add a `DW_OP_regx` operation to the expression.
+    ///
+    /// This may be emitted as a smaller equivalent operation.
+    pub fn op_reg(&mut self, register: Register) {
+        self.operations.push(Operation::Register(register));
+    }
+
+    /// Add a `DW_OP_implicit_value` operation to the expression.
+    pub fn op_implicit_value(&mut self, data: Box<[u8]>) {
+        self.operations.push(Operation::ImplicitValue(data));
+    }
+
+    /// Add a `DW_OP_implicit_pointer` operation to the expression.
+    pub fn op_implicit_pointer(&mut self, entry: Reference, byte_offset: i64) {
+        self.operations
+            .push(Operation::ImplicitPointer { entry, byte_offset });
+    }
+
+    /// Add a `DW_OP_piece` operation to the expression.
+    pub fn op_piece(&mut self, size_in_bytes: u64) {
+        self.operations.push(Operation::Piece { size_in_bytes });
+    }
+
+    /// Add a `DW_OP_bit_piece` operation to the expression.
+    pub fn op_bit_piece(&mut self, size_in_bits: u64, bit_offset: u64) {
+        self.operations.push(Operation::BitPiece {
+            size_in_bits,
+            bit_offset,
+        });
+    }
+
+    /// Add a `DW_OP_GNU_parameter_ref` operation to the expression.
```

Comment:
It would be nice to have some documentation about what those operations actually do as part of these doc comments.

Reply:
I think that should be documented on the operations instead (and this could link to that). I started doing that, but it ends up just copying the standard. While that might be useful, it is a low priority for me, since users should be referring to the standard anyway.